graylog2-server/src/main/java/org/graylog/events/context/EventDefinitionContextService.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.events.context;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import org.graylog.events.processor.EventDefinitionDto;
+import org.graylog.events.processor.EventProcessorExecutionJob;
+import org.graylog.scheduler.DBJobDefinitionService;
+import org.graylog.scheduler.DBJobTriggerService;
+import org.graylog.scheduler.JobDefinitionDto;
+import org.graylog.scheduler.JobTriggerData;
+import org.graylog.scheduler.JobTriggerDto;
+import org.graylog.scheduler.JobTriggerStatus;
+import org.joda.time.DateTime;
+
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class EventDefinitionContextService {
+    private final DBJobDefinitionService jobDefinitionService;
+    private final DBJobTriggerService jobTriggerService;
+
+    @Inject
+    public EventDefinitionContextService(DBJobDefinitionService jobDefinitionService, DBJobTriggerService jobTriggerService) {
+        this.jobDefinitionService = jobDefinitionService;
+        this.jobTriggerService = jobTriggerService;
+    }
+
+    public ImmutableMap<String, Object> contextFor(List<EventDefinitionDto> eventDefinitions) {
```

Comment:
Just an idea. If we use Serializable instead, it would be clearer that this is meant to
be sent out as a response..

```suggestion
    public ImmutableMap<String, Serializable> contextFor(List<EventDefinitionDto> eventDefinitions) {
```

Reply:
I would keep it as it is. We use `Map<String, Object>` for these types of return values all over the place. :smile:

Reply:
Fair enough